sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/policy/StorageBearerTokenChallengeAuthorizationPolicy.java

```diff
@@ -65,15 +70,41 @@ public Mono<Boolean> authorizeRequestOnChallenge(HttpPipelineCallContext context
         Map<String, String> challenges = extractChallengeAttributes(authHeader, BEARER_TOKEN_PREFIX);
 
         String scope = challenges.get("resource_id");
+        String authorization = challenges.get("authorization_uri");
+
         if (scope != null) {
             scope += DEFAULT_SCOPE;
             scopes = new String[] { scope };
             scopes = getScopes(context, scopes);
-            return setAuthorizationHeader(context, new TokenRequestContext().addScopes(scopes)).thenReturn(true);
         }
+
+        if (authorization != null) {
+            String tenantId = extractTenantIdFromUri(authorization);
+            TokenRequestContext tokenRequestContext = new TokenRequestContext().addScopes(scopes).setTenantId(tenantId);
+            return setAuthorizationHeader(context, tokenRequestContext).thenReturn(true);
+        }
+
+        if (scope != null) {
+            TokenRequestContext tokenRequestContext = new TokenRequestContext().addScopes(scopes);
+            return setAuthorizationHeader(context, tokenRequestContext).thenReturn(true);
+        }
+
         return Mono.just(false);
     }
 
+    String extractTenantIdFromUri(String uri) {
```

Comment:
Add some direct tests on this method for:

1. Invalid URIs
2. URIs without multiple path segments
3. Ensuring the right tenant ID is returned